### ISSUE TITLE
Tune optimization hyper-parameters on the validation dataset

### ISSUE THREAD

**Issue by accosmin** [accosmin/nano#188]
* modify apps/train to train multiple trials at once & save stats from these trials
* step 1: tune the optimization hyper-parameters on the validation set and configure the solver accordingly (no more automatic tuning for stochastic solvers)
* step2: for each trial: random initialization & optimize using the solver from step 1 & save the model

**Comment by accosmin** [comment]
* implement different learning rate schedules: https://cs231n.github.io/neural-networks-3/#baby
* e.g.: exponential or 1/t decay
